sockeye/arguments.py

```diff
@@ -529,9 +562,17 @@ def add_training_args(params):
                               help="Dropout probability for ConvolutionalEmbeddingEncoder. Default: %(default)s.")
 
     train_params.add_argument('--optimizer',
-                              default='adam',
-                              choices=['adam', 'sgd', 'rmsprop'],
+                              default=C.OPTIMIZER_ADAM,
+                              choices=C.OPTIMIZERS,
                               help='SGD update rule. Default: %(default)s.')
+    train_params.add_argument('--optimizer-params',
+                              type=simple_dict(),
+                              default=None,
+                              help='Additional optimizer params as dictionary. Format: key1:value1,key2:value2,...')
```

Comment:
I like the idea of a simple format instead of json here, as we anyway only need a flat list of keys and values. Just a as a ntoe: `key=value` would seem like the more natural format.

Reply:
One of the challenges is that shells like to do their own thing with characters that look like argument indicators.  Even trying to specify a negative number like `key:-1` causes problems.  This probably wants another iteration.